packages/perseus/src/components/graph.tsx

```diff
@@ -43,12 +45,12 @@ function numSteps(range: any, step: any) {
 }
 
 type Props = {
-    box: Readonly<[number, number]>;
+    box: [number, number];
     labels: ReadonlyArray<string>;
-    range: Readonly<[Coord, Coord]>;
-    step: Readonly<[number, number]>;
-    gridStep: Readonly<[number, number]>;
-    snapStep: Readonly<[number, number]>;
+    range: [Coord, Coord];
+    step: [number, number];
+    gridStep: [number, number];
+    snapStep: [number, number];
```

Comment:
For posterity, Ben and I paired through this and decided that the use of `Readonly<>` caused more issues with types than we thought it helped. So we removed these and we can review if we want to use `Readonly<T>` in the future once we've improved types in this area further. It might be an easier thing to do once we have more type coverage.

Reply:
Specifically, we couldn't figure out a way to type a readonly tuple that worked with type narrowing. The following code [does not typecheck](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcAUAPAXIgSgUwEMATBAGwE8AeAbTBAFsAjfAJwBpF7m2BdAPkQAfLoxasAlDm7jEAbwBQiRDGCI0AQVatCFAHQwAzlp0VMEifKXLErfFBCskGGgAZeiANSIXARl7WAL7Wdg5OPgrBQA):

```ts
function foo(x: Readonly<[number, number]> | number): number {
  if (Array.isArray(x)) {
    return x[0] + x[1]
  }
  return x
}
```